doc/design/ROS_COMPARISON.md

```diff
@@ -0,0 +1,37 @@
+# Navigation 2 System Overview
+
+## Comparison to Move Base in ROS
+AMCL and map_server were ported to ROS2 with minimal functional changes, but some refactoring.
+
+  * amcl -> nav2_amcl
+  * map_server -> nav2_map_server
+
+![Move Base 1](./move_base_compare_1.png)
+
+In addition, move_base itself has been split into multiple components:
+
+  * nav2_simple_navigator (replaces move_base)
+  * nav2_dijkstra_planner (replaces global_planner)
+  * nav2_controller_dwb (replaces local_planner)
+
+![Move Base 2](./move_base_compare_2.png)
+
+The *nav2_simple_navigator* replaces move_base at the top level, with a *Task* interface to call the global and local planners.
+
+* Note: the *Task* interface is a temporary proxy for ROS2 *Actions* which are not yet implemented. When *Actions* become available, the planners will be called through ROS2 *Actions*.
```

Comment:
A side note: I've read in a number of places the intention of using actionlib for many of these sub-components, in some cases I understand it, but not sure for local planners that makes sense (global planners I could see some applications that makes sense for). Actions I usually think of as processes that take a long time that shouldn't block something else that's happening for (see ROS answers note as well). A local planner should be running at 20+hz and would be a blocking call for the process trying to send them to the base. 

I also don't personally yet have a full comprehension on the delay and overhead that moving to DDS puts on breaking these things out, but especially as local planning is safety critical it should sit as close to the base/data as possible. 

https://answers.ros.org/question/11834/when-should-i-use-topics-vs-services-vs-actionlib-actions-vs-dynamic_reconfigure/

Reply:
@SteveMacenski I agree that we should cut over to a plug-in interface at the point that performance becomes an issue. In the specific case of the local planner, I think that the current nav2 idea is perhaps a bit different than the navigation 1 stack. In the nav2 case, the Action is "FollowPath" not, "GenerateLocalPlan." So, the FollowPath is a long-running action that would do the more frequent updates internally using plug-ins. 

Meanwhile, for the global planner, I think an action interface will be adequate and allow us to run the global planner relatively frequently. If it becomes and issue, we can revisit.

Reply:
@SteveMacenski - maybe we should open another issue for this discussion, but I'll give a brief 2 cents here. As @mjeronimo mentioned, the local planner can still run internally at its 20+hz rate. That's not impacted by this design. It will receive the plan through the Task/Action interface, but that is dependent on how often the Global planner is run, which is ~1 hz maximum. So even though there may be some additional latency between the time the new plan becomes available and the time it is received by the local planner, it should be in the 100ms range which should be OK for slow moving robots. Autonomous cars may need lower latency, but that's not our primary use case.

Reply:
For indoor slow moving robots, I would not consider 100ms to be acceptable. At 1 m/s that would be 10 cm before stopping to track a goal or turn around because of obstacles, people, etc. Many robots will move a little faster than 1 m/s at maximum. 

If the local planner is no longer being queried for individual velocities at 20hz, and its just separately spinning, using an action client definitely makes sense.

Reply:
Since actions aren't implemented yet, I don't think anybody can speak to the actual delay, right? In my (limited) understanding of ROS2 workflow, I imagine all of these components are composable and can run in the same process on most robots -- in which case, the actions should hopefully be fast (although the threading is possibly a nightmare from what I've read in the actions design guide).

Reply:
I don't think I've ever seen anyone run the global planner faster than 1hz -- so I think that assumption holds.

Reply:
I marked this as resolved within the PR, if we need to discuss more, let's open a separate issue. Thanks @mikeferguson and @SteveMacenski for the great feedback